File: csharp/main/Implementation/EnumerationNode.cs
using System;
using System.Collections.Generic;
using JetBrains.Annotations;

namespace Deltix.Luminary.Implementation
{
	internal class EnumerationNode : ItemNode
	{
		public EnumerationNode([NotNull] String name, [NotNull] String underlyingType,
			[CanBeNull, ItemNotNull] List<String> comments)
			: base(name, comments)
		{
			UnderlyingType = underlyingType;
		}

		public String UnderlyingType { get; }

		public List<EnumerationMemberNode> Members { get; } = new List<EnumerationMemberNode>();
	}
}

File: java/src/main/java/deltix/luminary/implementation/EnumerationNode.java
package deltix.luminary.implementation;

import java.util.ArrayList;
import java.util.List;

public class EnumerationNode extends ItemNode {
    private final String underlyingType;
    private final List<EnumerationMemberNode> members = new ArrayList<>();

    public EnumerationNode(String name, String underlyingType, List<String> comments) {
        super(name, comments);
        this.underlyingType = underlyingType;
    }

    public String getUnderlyingType() {
        return underlyingType;
    }

    public List<EnumerationMemberNode> getMembers() {
        return members;
    }
}
